core/src/main/java/com/oath/oak/OakAttachedReadBuffer.java

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright 2018 Oath Inc.
+ * Licensed under the terms of the Apache 2.0 license.
+ * Please see LICENSE file in the project root for terms.
+ */
+
+package com.oath.oak;
+
+import sun.nio.ch.DirectBuffer;
+
+import java.nio.ByteBuffer;
+import java.nio.ByteOrder;
+
+class OakAttachedReadBuffer implements OakReadBuffer, OakUnsafeDirectBuffer {
+
+    protected ByteBuffer bb;
+    protected int dataPos;
+    protected int dataLength;
+
+    OakAttachedReadBuffer(Slice s, ValueUtils valueOperator) {
+        bb = s.getByteBuffer();
```

Comment:
Is it a "big block" ByteBuffer?

Reply:
Yes. Not a sliced one.

Reply:
Depends what are you naming "sliced one". The thread's buffer's position and limit are set so it points on some slice... a portion of the "big block" buffer

Reply:
This is the per-thread buffer that is positioned to the key/value offset.